perun/utils/common/common_kit.py

```diff
@@ -437,4 +451,97 @@ def get_module(module_name: str) -> types.ModuleType:
     return MODULE_CACHE[module_name]
 
 
+def compact_convert_list_to_str(
+    number_list: list[int | float] | array.array[float] | array.array[int], float_precision: int = 2
+) -> list[str]:
+    """Converts list to list of compact strings
+
+    :param number_list: list of numbers
+    :param float_precision: float precision of the numbers
+    :return list of compact strings
+    """
+    return [compact_convert_num_to_str(num, 2) for num in number_list]
+
+
+def compact_convert_num_to_str(number: int | float, float_precision: int = 2) -> str:
+    """Converts the number to the smallest string possible
+
+    :param number: converted number
+    :param float_precision: float precision, i.e. number of decimal places in number
+    :return compact string
+    """
+    return str(to_compact_num(number, float_precision))
+
+
+def to_compact_num(number: int | float, float_precision: int = 2) -> int | float:
+    """Converts the number to the smallest string possible
+
+    :param number: converted number
+    :param float_precision: float precision, i.e. number of decimal places in number
+    :return compact num
+    """
+    if isinstance(number, int):
+        return number
+    elif number.is_integer():
+        return int(number)
+    else:
+        return round(number, float_precision)
+
+
+@contextlib.contextmanager
+def disposable_resources(disposable: Any) -> Any:
+    """Helper context manager that disposes of the passed object
+
+    :param disposable: object that should be disposed immediately
+    :return: object
+    """
+    try:
+        yield disposable
+    except Exception:
+        # Re-raise the encountered exception
+        raise
+    finally:
+        del disposable
+        gc.collect()
```

Comment:
Neat, just curious, isn't there a heavy performance penalty for calling GC? Maybe the docstring should mention the potential overhead so that others don't overuse it.

Reply:
Good Idea to extend it with docstring. Just little anecdote: I worked with some huge json files that I was converting to pandas.DataFrame and then concatenating into single one; the explicit call to gc actually made it feasiable, as it simply refused to collect the garbage in between the loading (it was in for loop). So manual GC can definitely help, but needs some tweaks.